diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt
index cb31893..d766760 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt
@@ -90,7 +90,10 @@ class SafeAreaView(
     }
 
     fun onInterfaceInsetsChange(newInterfaceInsets: EdgeInsets) {
-        if (newInterfaceInsets != currentInterfaceInsets) {
+        if (newInterfaceInsets.top != currentInterfaceInsets.top ||
+            newInterfaceInsets.right != currentInterfaceInsets.right ||
+            newInterfaceInsets.bottom != currentInterfaceInsets.bottom ||
+            newInterfaceInsets.left != currentInterfaceInsets.left) {
             currentInterfaceInsets = newInterfaceInsets
 
             if (insetType.containsInterface()) {
@@ -106,8 +109,12 @@ class SafeAreaView(
         val newSystemInsets =
             insets.getInsets(WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.displayCutout())
 
-        if (newSystemInsets != currentSystemInsets) {
-            currentSystemInsets = EdgeInsets.fromInsets(newSystemInsets)
+        val newSystemEdgeInsets = EdgeInsets.fromInsets(newSystemInsets)
+        if (newSystemEdgeInsets.top != currentSystemInsets.top ||
+            newSystemEdgeInsets.right != currentSystemInsets.right ||
+            newSystemEdgeInsets.bottom != currentSystemInsets.bottom ||
+            newSystemEdgeInsets.left != currentSystemInsets.left) {
+            currentSystemInsets = newSystemEdgeInsets
 
             if (insetType.containsSystem()) {
                 needsInsetsUpdate = true
